ui: Add reviewer view

zhangvi7 opened the pull request:
## Context
- Add reviewer view, where assigned reviewers can now approve/reject a query execution that is pending review.
- Scenarios:
   - Feature completely disabled in public config: we remove toggle in admin settings, and hide reviews icon on left side bar
   - Feature enabled but not used: We only show reviewer view on an execution if a review exists
   - Feature enabled and review requested: Show reviewer view with historical review context for approved/rejected queries, or approve/reject button for pending review queries

## Changes
- Reviewer UI
- Approve/Reject: endpoints, notifications, redux (fetch review along with query execution in QueryView.tsx)

## Test Plan

Demo:

https://github.com/user-attachments/assets/91f4f702-f1a2-4f5c-8ba8-c2757ba002ac

Assigned Reviewer:
<img width="1301" alt="image" src="https://github.com/user-attachments/assets/017ecc8c-d2d4-48a5-94f1-a02a41c3ac44" />

Not Reviewer, Pending Review: 
![Screenshot 2025-01-24 at 7 24 51 PM](https://github.com/user-attachments/assets/9e091ecb-bf79-4b07-83d5-37d5b5376478)

Approved Review:
<img width="1321" alt="image" src="https://github.com/user-attachments/assets/755a1a70-3e14-43a5-ac76-d97693935530" />

Rejected Review:
<img width="1684" alt="image" src="https://github.com/user-attachments/assets/759783c2-8ae1-4d33-baf0-1d4401eaba50" />

zhangvi7 committed:
ui: Add reviewer view

jczhong84 commented on querybook/webapp/components/QueryViewReview/QueryViewReview.tsx:
```diff
@@ -0,0 +1,368 @@
+        <div className="review-actions">
+            <div className="action-buttons">
+                <Button
+                    title="Request Changes"
```
why use "Request Changes" instead of "Reject"? 
"Request Changes" seems to be the same review request will be sent back again, while the current design is a new request will be created iirc?

jczhong84 commented on querybook/webapp/components/QueryViewReview/QueryViewReview.tsx:
```diff
@@ -0,0 +1,368 @@
+                />
+            </div>
+
+            {showApproveModal && (
```
instead of displaying a modal over modal for the approval and rejection,  would it be better to not use a modal for the request detail, but display it on the right side directly?

zhangvi7 replied:
I saw few other instances in Querybook where we have a nest modal for confirmation, so I don't think it's big issue if just for confirmation. Also the query view won't open as modal if opened execution url directly i.e. in https://querybook.pinadmin.com/data/query_execution/15430515/ what do you think? @jczhong84

zhangvi7 replied:
<img width="1680" alt="image" src="https://github.com/user-attachments/assets/4662f7eb-b9ec-4518-b53f-37792a64765a" />

jczhong84 replied:
yeah, I think it looks much better. The reason for other instances like tables, snippet, they are actually for helping the query authoring, so we'd like the datadoc is still there. While for query review, it's not related to the current datadoc. 

if it's an easy change, we can include it here. otherwise we can do it in a separately.

zhangvi7 replied:
Changed so onClick of review item redirects instead of open a modal

Commit:
remove review item modal

Commit:
fix tab styling

zhangvi7 commented:
Changes:
- Reduced font size on empty review text 
- Moved submit button to bottom right
- Tabs resize on panel change
- Updated modal text and styling

![Screenshot 2025-01-30 at 12 37 42 PM](https://github.com/user-attachments/assets/324e2392-f37a-4c99-bd84-4b96a3eb46ce)
